### PR TITLE
Upgrade PyTorch to 1.8.0

### PR DESCRIPTION
**Pull request by goswamig**
## Description ##
Upgrading PyTorch to v1.8.0

```
$ ./gradlew compileJNI 

> Task :pytorch:pytorch-native:compileJNI
~/djl/pytorch/pytorch-native ~/djl/pytorch/pytorch-native
-- The C compiler identification is GNU 5.4.0
-- The CXX compiler identification is GNU 5.4.0
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /usr/bin/cc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /usr/bin/c++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
Building djl_torch with the host...
-- Looking for pthread.h
-- Looking for pthread.h - found
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD - Failed
-- Looking for pthread_create in pthreads
-- Looking for pthread_create in pthreads - not found
-- Looking for pthread_create in pthread
-- Looking for pthread_create in pthread - found
-- Found Threads: TRUE  
-- Found Torch: /home/ubuntu/djl/pytorch/pytorch-native/libtorch/lib/libtorch.so  
-- Found JNI: /usr/lib/jvm/java-8-openjdk-amd64/jre/lib/amd64/libjawt.so  
-- Configuring done
-- Generating done
-- Build files have been written to: /home/ubuntu/djl/pytorch/pytorch-native/build
Scanning dependencies of target djl_torch
[  6%] Building CXX object CMakeFiles/djl_torch.dir/src/main/native/ai_djl_pytorch_jni_PyTorchLibrary_system.cc.o
[ 12%] Building CXX object CMakeFiles/djl_torch.dir/src/main/native/ai_djl_pytorch_jni_PyTorchLibrary_nn_functional.cc.o
[ 18%] Building CXX object CMakeFiles/djl_torch.dir/src/main/native/ai_djl_pytorch_jni_PyTorchLibrary_inference.cc.o
[ 25%] Building CXX object CMakeFiles/djl_torch.dir/src/main/native/ai_djl_pytorch_jni_PyTorchLibrary_torch_comparison.cc.o
[ 31%] Building CXX object CMakeFiles/djl_torch.dir/src/main/native/ai_djl_pytorch_jni_PyTorchLibrary_torch_pointwise.cc.o
[ 37%] Building CXX object CMakeFiles/djl_torch.dir/src/main/native/ai_djl_pytorch_jni_PyTorchLibrary_optim.cc.o
[ 43%] Building CXX object CMakeFiles/djl_torch.dir/src/main/native/ai_djl_pytorch_jni_PyTorchLibrary_torch_creation.cc.o
[ 50%] Building CXX object CMakeFiles/djl_torch.dir/src/main/native/ai_djl_pytorch_jni_PyTorchLibrary_ivalue.cc.o
[ 56%] Building CXX object CMakeFiles/djl_torch.dir/src/main/native/ai_djl_pytorch_jni_PyTorchLibrary_torch_random_sampling.cc.o
[ 62%] Building CXX object CMakeFiles/djl_torch.dir/src/main/native/ai_djl_pytorch_jni_PyTorchLibrary_torch_other.cc.o
[ 68%] Building CXX object CMakeFiles/djl_torch.dir/src/main/native/ai_djl_pytorch_jni_PyTorchLibrary_torch_isjm.cc.o
[ 75%] Building CXX object CMakeFiles/djl_torch.dir/src/main/native/ai_djl_pytorch_jni_PyTorchLibrary_torch_reduction.cc.o
[ 81%] Building CXX object CMakeFiles/djl_torch.dir/src/main/native/ai_djl_pytorch_jni_PyTorchLibrary_tensor.cc.o
[ 87%] Building CXX object CMakeFiles/djl_torch.dir/src/main/native/ai_djl_pytorch_jni_cache.cc.o
[ 93%] Building CXX object CMakeFiles/djl_torch.dir/src/main/native/djl_pytorch_jni_log.cc.o
[100%] Linking CXX shared library libdjl_torch.so
[100%] Built target djl_torch
~/djl/pytorch/pytorch-native

Deprecated Gradle features were used in this build, making it incompatible with Gradle 7.0.
Use '--warning-mode all' to show the individual deprecation warnings.
See https://docs.gradle.org/6.7.1/userguide/command_line_interface.html#sec:command_line_warnings

BUILD SUCCESSFUL in 12s
1 actionable task: 1 executed
```

### CODE REVIEW

**Review by stu1130 — APPROVED** [review]

**Comment by lanking520** [comment]
Currently I added some placeholder JNI binaries in 1.8.0 `jnilib`. This should mitigate download issue. This PR will fail on the issue for PyTorch native auto not found issue. I would propose we merge it anyway and proceed with the following actions:

1. run the native publish workflow to build snapshot
2. run the JNI publish workflow to publish the updated JNI (possibly bug fixes if some issues are found).

File: .github/workflows/native_jni_s3_pytorch.yml [hunk 1]
```diff
@@ -140,30 +140,26 @@ jobs:
       - name: Instal CUDA
```
Comment (lanking520) [inline .github/workflows/native_jni_s3_pytorch.yml:140]: Let's verify if the JNI build on all OS is succeed

**Review by lanking520 — CHANGES_REQUESTED** [review]
Will remove once the JNI build verification complete

**Review by stu1130 — APPROVED** [review]

**Comment by stu1130** [comment]
remove the -SNAPSHOT, we don't have 1.8.0-SNAPSHOT yet

**Comment by lanking520** [comment]
We don't have 1.8.0 either. 1.8.0-SNAPSHOT will come first

**Comment by stu1130** [comment]
Let's still use the 1.7.1 in gradle.properties and then we can merge the PR and build the JNI

**Comment by lanking520** [comment]
> Let's still use the 1.7.1 in gradle.properties and then we can merge the PR and build the JNI

Why this makes a difference? It will break anyway even if we switch to 1.7.1

**Review by frankfliu — APPROVED** [review]

File: .github/workflows/native_jni_s3_pytorch.yml [hunk 2]
```diff
@@ -140,30 +140,26 @@ jobs:
...
           mkdir cuda
           unzip.exe cudnn.zip
-          cp.exe -a cuda/include cuda/lib cuda/bin "C:/Program Files/NVIDIA GPU Computing Toolkit/CUDA/v10.1/"
           cp.exe -a cuda/include cuda/lib cuda/bin "C:/Program Files/NVIDIA GPU Computing Toolkit/CUDA/v10.2/"
           rm.exe -Rf cuda102.exe cuda.exe cudnn.zip cuda
       - name: Install CUDA 11
         shell: cmd
         run: |
-          curl.exe -L http://developer.download.nvidia.com/compute/cuda/11.0.2/local_installers/cuda_11.0.2_451.48_win10.exe -o cuda.exe
+          curl.exe -L https://developer.download.nvidia.com/compute/cuda/11.1.0/local_installers/cuda_11.1.0_456.43_win10.exe -o cuda.exe
```
Comment (lanking520) [inline .github/workflows/native_jni_s3_pytorch.yml:153]: Just to confirm, have you verified the windows build here? Any link to the successful JNI build Action?

Comment (goswamig) [reply .github/workflows/native_jni_s3_pytorch.yml:153]: not yet...can I take windows build in separate PR ?

**Review by lanking520 — APPROVED** [review]

File: .github/workflows/native_jni_s3_pytorch.yml [hunk 3]
```diff
@@ -140,30 +140,26 @@ jobs:
       - name: Instal CUDA
         shell: cmd
         run: |
-          curl.exe -L http://developer.download.nvidia.com/compute/cuda/10.1/Prod/local_installers/cuda_10.1.243_426.00_windows.exe -o cuda.exe
           curl.exe -L http://developer.download.nvidia.com/compute/cuda/10.2/Prod/local_installers/cuda_10.2.89_441.22_windows.exe -o cuda102.exe
           curl.exe -L https://developer.download.nvidia.com/compute/redist/cudnn/v7.6.4/cudnn-10.1-windows7-x64-v7.6.4.38.zip -o cudnn.zip
```
Comment (goswamig) [inline .github/workflows/native_jni_s3_pytorch.yml:144]: @lanking520 is this right cudnn-10.1 ?

File: .github/workflows/native_jni_s3_pytorch.yml [hunk 4]
```diff
@@ -140,30 +140,26 @@ jobs:
...
         shell: cmd
         run: |
           call "C:\Program Files (x86)\Microsoft Visual Studio\2019\Enterprise\VC\Auxiliary\Build\vcvarsall.bat" amd64
           set "CUDA_PATH=%ProgramFiles%\NVIDIA GPU Computing Toolkit\CUDA\v10.2"
           set "CUDA_PATH_V10_2=%ProgramFiles%\NVIDIA GPU Computing Toolkit\CUDA\v10.2"
-          set "CUDA_PATH_V10_1=%ProgramFiles%\NVIDIA GPU Computing Toolkit\CUDA\v10.1"
-          set "CUDA_PATH_V11_0=%ProgramFiles%\NVIDIA GPU Computing Toolkit\CUDA\v11.0"
-          set "PATH=%ProgramFiles%\NVIDIA GPU Computing Toolkit\CUDA\v10.1\bin;%ProgramFiles%\NVIDIA GPU Computing Toolkit\CUDA\v10.1\libnvvp;%PATH%"
+          set "CUDA_PATH_V11_1=%ProgramFiles%\NVIDIA GPU Computing Toolkit\CUDA\v11.1"
+          set "PATH=%ProgramFiles%\NVIDIA GPU Computing Toolkit\CUDA\v10.2\bin;%ProgramFiles%\NVIDIA GPU Computing Toolkit\CUDA\v10.2\libnvvp;%PATH%"
```
Comment (goswamig) [inline .github/workflows/native_jni_s3_pytorch.yml:162]: @lanking520 I think PATH is wrong here right, we are still pointing it to cuda-10.2 ?

File: .github/workflows/native_jni_s3_pytorch.yml (hunk 3 above)
Comment (lanking520) [reply .github/workflows/native_jni_s3_pytorch.yml:144]: It should be generally applicable for 10.2, you can try to switch to 10.2

File: .github/workflows/native_jni_s3_pytorch.yml (hunk 4 above)
Comment (lanking520) [reply .github/workflows/native_jni_s3_pytorch.yml:162]: Yes, still pointing to 10.2 works here

**Pull request merged**